### PR TITLE
#1123 - Update enhancements

### PR DESCRIPTION
**Pull request by kvanhijf**

### CODE REVIEW

**Comment by kvanhijf** [comment]
https://github.com/openvstorage/framework/pull/1141
https://github.com/openvstorage/alba-asdmanager/pull/165
https://github.com/openvstorage/framework-alba-plugin/pull/278

File: source/app/api.py [hunk 1]
```diff
@@ -258,38 +266,35 @@ def asd_delete(disk_id, asd_id):
...
+    @post('/update/execute')
+    def update():
         """
         Execute an update
-        :param status: Current status of the update
-        :type status: str
         """
         with file_mutex('package_update'):
-            return UpdateController.execute_update(status)
+            return SDMUpdateController.update()
```
Comment (khenderick) [inline source/app/api.py]: Can you track progress in any way? If this call takes too long the client will timeout, so you need some async system.

File: source/controllers/update.py [hunk 2]
```diff
@@ -18,108 +18,110 @@
...
-        try:
-            UpdateController.update_package_cache()
-            sdm_package_info = UpdateController.get_package_information(package_name=UpdateController.PACKAGE_NAME)
-        except CalledProcessError:
-            return {'status': 'started'}
+        package_info = {}
+        default_entry = {'candidate': None,
+                         'installed': None,
+                         'services_to_restart': []}
+        #                     component: package_name: services_with_run_file
```
Comment (khenderick) [inline source/controllers/update.py]: Is this still required?

File: source/tools/packages/debian.py [hunk 3]
```diff
@@ -18,117 +18,118 @@
...
-                    if client.file_exists('/opt/OpenvStorage/run/{0}.version'.format(service)):
-                        running_version = client.file_read('/opt/OpenvStorage/run/{0}.version'.format(service)).strip()
-                        if running_version != candidate:
-                            update_info['services'].append(service)
-                            update_info['version'] = candidate
-        return update_info
+        if installed == candidate:
+            return
+
+        command = 'aptdcon --hide-terminal --allow-unauthenticated --install {0}'.format(package_name)
```
Comment (khenderick) [inline source/tools/packages/debian.py]: Escaping (assume you need a sync on the framework's fixes)

File: source/app/api.py (hunk 1 above)
Comment (kvanhijf) [inline source/app/api.py]: Timeout has been set on 5 minutes
And progress can be tracked by checking the logging i guess

File: source/tools/packages/debian.py [hunk 4]
```diff
@@ -46,7 +46,7 @@ def get_installed_versions(client=None, package_names=None):
         if package_names is None:
             package_names = DebianPackage.SDM_PACKAGE_NAMES
         for package_name in package_names:
-            command = "dpkg -s {0} | grep Version | awk '{{print $2}}'".format(package_name)
+            command = "dpkg -s {0} | grep Version | awk '{{print $2}}'".format(package_name.replace(r"'", r"'\''"))
```
Comment (khenderick) [inline source/tools/packages/debian.py]: Same as in the framework

**Pull request merged**